.github/workflows/build-and-cache.yaml

```diff
@@ -7,11 +7,33 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  nix:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v4
+
+      - name: Install nix
+        uses: cachix/install-nix-action@v27
+        with:
+          install_url: https://releases.nixos.org/nix/nix-2.22.1/install
+
+      - name: Check Nix formatting
+        run: nix fmt . -- --check
+
+      - name: Check Nix flake
+        run: nix flake check --all-systems
+
   build:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, macos-13]
-        package: [holochain, lair-keystore, hc-launch, hc-scaffold]
+        packages:
+          - [holochain, hc, hc-run-local-services, hc-sandbox, hcterm]
```

Comment:
Grouped because they all build the same derivation - so even with a cache it will de-duplicate work to run these on the same machine